### ISSUE TITLE
 Test label leakage problem

### ISSUE THREAD

**Issue by Mdlw26** [LARS-research/CLGNN#1]
Hi,

I want to bring to your attention to the test label leakage problem occurs during the calculation of the loss function (function 'cl_forward_and_compute_loss' in [torch_train_model.py](https://github.com/LARS-research/CLGNN/blob/2d3c75fd10154889dccbead076472705cd76e1f0/rphgnn/layers/torch_train_model.py#L359)).

Specifically, the function first calculates loss for all training samples (Line 365-367), followed by selecting top-k training samples with the smallest loss (the key idea of curriculum learning) (Line 372-378).
The label leakage happens in Lines 380-381,  where the selected training samples are used to calculate a new loss.   The correct code here should be `logits[batch_train_mask][selected_idx],` but the original code uses `logits[selected_idx]`, which neglect the training mask, thus inevitably introducing some test samples to calculate the loss.
![image-2](https://github.com/user-attachments/assets/cbc0a1ee-721b-41cd-a3b3-9fe51e7f0c98)

I reran the code with the corrected Line 380:
`ce_loss_list = [self.loss_func(logits[batch_train_mask][selected_idx], batch_y[batch_train_mask][selected_idx] for logits in logits_list].`
and get similar results (valid_acc = 0.5902, test_acc = 0.5725) compared to the baseline model RpHGNN (valid_acc = 0.5973, test_acc = 0.5773).

Thank you for your attention to this matter.

**Comment by Mdlw26** [comment]
@weihua916

**Comment by wongzhenhao** [comment]
> Hi,
> 
> I want to bring to your attention to the test label leakage problem occurs during the calculation of the loss function (function 'cl_forward_and_compute_loss' in [torch_train_model.py](https://github.com/LARS-research/CLGNN/blob/2d3c75fd10154889dccbead076472705cd76e1f0/rphgnn/layers/torch_train_model.py#L359)).
> 
> Specifically, the function first calculates loss for all training samples (Line 365-367), followed by selecting top-k training samples with the smallest loss (the key idea of curriculum learning) (Line 372-378). The label leakage happens in Lines 380-381,  where the selected training samples are used to calculate a new loss.   The correct code here should be `logits[batch_train_mask][selected_idx],` but the original code uses `logits[selected_idx]`, which neglect the training mask, thus inevitably introducing some test samples to calculate the loss. ![image-2](https://private-user-images.githubusercontent.com/175461497/348444421-cbc0a1ee-721b-41cd-a3b3-9fe51e7f0c98.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA4ODQ5MDcsIm5iZiI6MTcyMDg4NDYwNywicGF0aCI6Ii8xNzU0NjE0OTcvMzQ4NDQ0NDIxLWNiYzBhMWVlLTcyMWItNDFjZC1hM2IzLTlmZTUxZTdmMGM5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxM1QxNTMwMDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZGQzN2Y3NDNlZjhlYTBhMGQ0YWE0NzA5NGZjZWQxZDc1ZWE0MTE0NzJiMWQxNTJlMDAzZjk1NGRiMzQ4MzljJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.SbTX3SrV77IINRU5d3ymN9qgtI9gtLhBSdMgYzLHFNg)
> 
> I reran the code with the corrected Line 380: `ce_loss_list = [self.loss_func(logits[batch_train_mask][selected_idx], batch_y[batch_train_mask][selected_idx] for logits in logits_list].` and get similar results (valid_acc = 0.5902, test_acc = 0.5725) compared to the baseline model RpHGNN (valid_acc = 0.5973, test_acc = 0.5773).
> 
> Thank you for your attention to this matter.

We will further refine and analyze the methods, aiming to provide new updates in September.

**Comment by weihua916** [comment]
@wongzhenhao Hi! This is OGB team. thanks for investigating.
If possible, could you investigate ASAP, since while 20% absolute improvement is truly impressive, it could mislead the community if it is not true. If it takes more than 2 weeks to investigate, we would withhold your submission from the leaderboard until your September update.

**Comment by wongzhenhao** [comment]
noted with thanks!

**Comment by weihua916** [comment]
just to clarify -- are you getting back to us with your investigation, or do you want us to remove your leaderboard submission?

**Comment by wongzhenhao** [comment]
We will continue to investigate this issue, thank you.

**Comment by quanmingyao** [comment]
> just to clarify -- are you getting back to us with your investigation, or do you want us to remove your leaderboard submission?

Thanks for helping us to deal with this issue :)

**Comment by quanmingyao** [comment]
> just to clarify -- are you getting back to us with your investigation, or do you want us to remove your leaderboard submission?

Yes, please.

**Comment by weihua916** [comment]
Removing this from OGB leaderboard for now, as there is no further response from the author.

**Comment by Skyorca** [comment]
OMG...
